myadmin-service-impls/student-services/src/main/java/za/ac/unisa/myadmin/student/services/controllers/StudentRestServiceImpl.java

```diff
@@ -0,0 +1,71 @@
+import org.springframework.web.bind.annotation.RestController;
+import za.ac.unisa.myadmin.common.exceptions.DoesNotExistException;
+import za.ac.unisa.myadmin.common.exceptions.InvalidParameterException;
+import za.ac.unisa.myadmin.common.exceptions.MissingParameterException;
+import za.ac.unisa.myadmin.common.exceptions.OperationFailedException;
+import za.ac.unisa.myadmin.student.services.dto.StudentInfo;
+import za.ac.unisa.myadmin.student.services.student.StudentService;
+
+import java.util.List;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+
+@RestController
+@RequestMapping({"/studentservices"})
+public class StudentRestServiceImpl {
+
+	@Autowired
+	@Qualifier("StudentService")
+	private StudentService studentService;
+
+	/**
+	 * Forgotten Student Number resource
+	 *
+	 * @param studentRequest
+	 * @return Valid student otherwise error response.
+	 * @throws OperationFailedException
+	 * @throws MissingParameterException
+	 * @throws InvalidParameterException
+	 * @throws DoesNotExistException
+	 */
+	@PostMapping(path = "/student/find", consumes = APPLICATION_JSON_VALUE, produces = APPLICATION_JSON_VALUE)
+	public StudentInfo requestStudentDetails(@RequestBody StudentInfo studentRequest) throws OperationFailedException, MissingParameterException, InvalidParameterException, DoesNotExistException {
+		List<StudentInfo> returnStudentInfo;
+		//Validate required request params
+		if (!StringUtils.hasText(studentRequest.getSurname())) {
+			throw new OperationFailedException("Please enter your surname.");
+		}
+		if (!StringUtils.hasText(studentRequest.getFirstNames())) {
+			throw new OperationFailedException("Please enter your first name(s).");
+		}
+		if (studentRequest.getDateOfBirth() == null) {
+			throw new OperationFailedException("Please enter your date of birth.");
+		}
+
+		if (!StringUtils.hasText(studentRequest.getIdentityNumber()) && !StringUtils.hasText(studentRequest.getPassportNumber())) {
+			returnStudentInfo = studentService.getStudentsBySurnameAndFirstNamesAndBirthDate(StringUtils.trimWhitespace(studentRequest.getSurname()), StringUtils.trimWhitespace(studentRequest.getFirstNames()), studentRequest.getDateOfBirth());
+		} else if (StringUtils.hasText(studentRequest.getIdentityNumber())) {
+			returnStudentInfo = studentService.getStudentsBySurnameAndFirstNamesAndBirthDateAndIdNumber(StringUtils.trimWhitespace(studentRequest.getSurname()), StringUtils.trimWhitespace(studentRequest.getFirstNames()), studentRequest.getDateOfBirth(), StringUtils.trimWhitespace(studentRequest.getIdentityNumber()));
+		} else {
+			returnStudentInfo = studentService.getStudentsBySurnameAndFirstNamesAndBirthDateAndPassportNumber(StringUtils.trimWhitespace(studentRequest.getSurname()), StringUtils.trimWhitespace(studentRequest.getFirstNames()), studentRequest.getDateOfBirth(), StringUtils.trimWhitespace(studentRequest.getPassportNumber()));
+		}
+		if (CollectionUtils.isEmpty(returnStudentInfo)) {
```

Comment:
This errors must not be thrown inside the rest layer but rather be thrown as a DoesNotExistException by the underlying service itself.